cmd/zfs/zfs_main.c

```diff
@@ -2537,14 +2572,16 @@ userspace_cb(void *arg, const char *domain, uid_t rid, uint64_t space)
 				if ((g = getgrgid(rid)) != NULL)
 					name = g->gr_name;
 			}
-		} else {
+		} else if (zfs_prop_is_user(prop)) {
 			type = USTYPE_PSX_USR;
 			if (!cb->cb_numname) {
 				struct passwd *p;
 
 				if ((p = getpwuid(rid)) != NULL)
 					name = p->pw_name;
 			}
+		} else {
+			type = USTYPE_PROJ;
```

Comment:
Support needs to be added to resolved a projid in the [/etc/projid](http://man7.org/linux/man-pages/man5/projid.5.html) file if it exists.  This will allow the `-n` option to be re-enabled, and you'll want to update this section of the `zfs(8)` man page to mention projects.

```
       -n  Print numeric ID instead of user/group/project name.
```

Reply:
> I'm still not entirely clear on this. I see that in the zfs_sa_upgrade() case PROJID will be 
> placed after LINKS since sa_replace_all_by_template() preserves the passed order. 
> However, I don't see where this ordering is made the preferred default layout for new 
> filesystems. Can you point me to the code which ensures this is true.

You mean the logic for zfs_mknode()? It is also after the LINKS.

Reply:
Yup that's the one.  Thanks.